### DISCUSSION TITLE
[Feature] Full management from within albums

### DISCUSSION THREAD

**Discussion by bcsteeve** [immich-app/immich#2997] [Feature Request]
### The feature

When in the main view, I can delete a file (for example).

When in an album, I can remove it from the album, which makes sense, but I can't delete the file.  Or archive, or add to a different album, etc.

I think it would be useful to have the same full access inside an album as I do outside of the album.

Reason/use-case:  Large collections are difficult to manage.  I may know that I put IMG_001.jpg into my "cats" album, but otherwise have no idea where it is.  Maybe I really want to delete that stupid cat from my life.  I can find it in the album, but I can't do anything with it from there.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by jrasm91** [comment]
Internally we reuse an action bar component in almost all of the pages, which has helped make it more consistent what actions are available where. However, albums don't use it yet. They are quite complicated and that needs to be re worked. I think after that is done and they use the same action bar component what you are asking would be easy to add and it would be more consistent with the rest of the app.

**Comment by bcsteeve** [comment]
A few days later of playing with Immich, I think I should change this request to simply be "albums consistent with Photos".  

In album view:

 - less functionality (as discussed above)
 - Thumbnails are worse.  They don't respect the nice landscape vs. portrait format of the main Photos page.  This **is** a feature of Google Photos.
 - Date display. I argue that an album should be sorted and segmented by date, just like the main Photos page.  This **is not** a feature of Google Photos, but that seems to be an oversight.  It should at least be an option.  Conceivably, some albums may cover decades.  Note:  allowing for nested albums would also work, I think, as people could choose to divide the album up however they wish.
 - Navigation pane.  Again, not a current feature of Google Photos, but why does the navigation pane go away when in album view?  That's just more clicks (more clicks almost necessarily means bad UX).

Wouldn't it be far easier to have the album view re-use the main photos code, but just accessing a subset of the photos (ie. whatever is marked as belonging to that album)?  Better UX, easier to implement... win/win?

**Reply by alextran1502** [reply]
Please follow #2464 for update. We all know the performance advantages of the timeline view. We got some refactoring work to do  for it to happens though 😊

**Reply by alextran1502** [reply]
For any UI feedback, a screenshot is appreciated

**Comment by bcsteeve** [comment]
The ability to archive from within an album would really help me solve a problem I'm having right now... in case that's not too hard to add :)

**Comment by jrasm91** [comment]
Tracking in #3607
